docs/developer-guide.md

```diff
@@ -1,5 +1,9 @@
 # Developer Guide
```

Comment:
Spark .NET Developer Guide? (since we want to differentiate from an Application Developer Guide)